### ISSUE TITLE
Two words split by code block in MDX

### ISSUE THREAD

**Issue by verhovsky** [jdkato/txtlint#1052]
### Check for existing issues

- [x] Completed

### Environment

- macOS
- Homebrew
- `vale version 3.13.0`

### Describe the bug / provide steps to reproduce it

This file:

```mdx
in `$VARIABLE` for.

information
```

when parsed as MDX with this .vale.ini

```ini
SkippedScopes = code

[formats]
mdx = md

[*.mdx]
BasedOnStyles = Vale
```

Gives this error:

```
 test.mdx
 3:1  error  Did you really mean 'infor'?  Vale.Spelling 

✖ 1 error, 0 warnings and 0 suggestions in 1 file.
```

There should be no error because "in" "for" and "information" are all normal words. For some reason having "in" and "for" separated with a `code` block and then followed by a word that uses both words causes an error on that word (notice the error is on line 3, not 1).

**Comment by colinloretz** [comment]
I am seeing this as well across multiple words where it is detecting tokens inside full words.


Specifically for the `infor` one, the line is:

```markdown
The interaction payload will also include a new `attachment_size_limit` key that specifies the maximum allowed attachment size. This limit may be higher than the default attachment size limit, depending on the guild's boost status or the invoking user's Nitro status.

For more information, check out [our documentation on file uploads](/developers/docs/reference#uploading-files).
```

When tested in isolation, the same string passes fine:

```sh
echo "For more information, check out [our documentation on file uploads](/developers/docs/reference#uploading-files)." | vale --ext=.mdx
```

Other examples of false positives:

```sh
 developers/docs/developer-tools/game-sdk.mdx
 157:238  error  Did you really mean 'inin'?    Vale.Spelling 
 274:27   error  Did you really mean 'callin'?  Vale.Spelling 


 developers/docs/events/gateway.mdx
 304:184  error  Did you really mean 'passin'?  Vale.Spelling 
 609:39   error  Did you really mean 'asin'?    Vale.Spelling 


 developers/docs/interactions/overview.mdx
 162:239  error  Did you really mean 'ato'?  Vale.Spelling 


 developers/docs/resources/channel.mdx
 553:48  error  Did you really mean 'bein'?  Vale.Spelling 

 developers/docs/tutorials/developing-a-user-installable-app.mdx
 358:220  error  Did you really mean 'inin'?  Vale.Spelling 

 developers/docs/change-log.mdx
 745:12  error  Did you really mean 'infor'?  Vale.Spelling 
```
